File: client/src/app/_models/aircraft.ts
export class Aircraft {
    id: number;
    size: Size;
    maxSeats: number;
    maxLuggageSize: number;


  constructor(size: Size, maxSeats: number, maxLuggageSize: number) {
    this.size = size;
    this.maxSeats = maxSeats;
    this.maxLuggageSize = maxLuggageSize;
  }
}

File: reservetickets/src/main/java/ticketservice/model/Aircraft.java
package ticketservice.model;

import javax.persistence.*;

import lombok.Getter;
import lombok.Setter;

import java.util.LinkedHashSet;
import java.util.Set;

@Getter
@Setter
@Entity
public class Aircraft {

	@Id
	@GeneratedValue
	private Integer id;

	private Size size;
	private Integer maxSeats;
	private Integer maxLuggageSize;


	@OneToMany(mappedBy="aircraft")
	private Set<Flight> flights;

	public Aircraft() {
		size = Size.MEDIUM;
		maxSeats = 0;
		maxLuggageSize = 0;
	}

	public Aircraft(Size size, Integer maxSeats, Integer maxLuggageSize) {
		this.flights = new LinkedHashSet<>();
		this.size = size;
		this.maxSeats = maxSeats;
		this.maxLuggageSize = maxLuggageSize;
	}

}
